### ISSUE TITLE
IE8 fix

### ISSUE THREAD

**Issue by Robhub** [p3drosola/Backbone.VirtualCollection#60]
Hello, I had problems on IE8, here are the changes I did to make it work
- in _rebuildIndex: check that this.models[idx].off is a function before calling it
- in buildFilter: replace Object.keys by _.keys

**Comment by p3drosola** [comment]
Looks good! I'd be happy to merge that if you make a PR.

In what situation could `this.models[idx].off` not be a function?

**Comment by mayang** [comment]
I had the _rebuildIndex problem on Chrome too. It's because the idx in for (idx in this.models) can be a 'last' and this.models['last'] is a method which doesn't have 'off'. I fixed it by switching to a forEach and checking for this.models existence

**Comment by p3drosola** [comment]
What? I don't understand what you're describing.
```
for (i in [1, 2, 3]) {
  console.log(i);
}
```

how is `i` ever `"last"`?

**Comment by mayang** [comment]
Hm. When I do it in the console it doesn't do last but when I do it running the app I'm using Virtual Collection i do get 'last' 

```
for(idx in [1,2,3]) { console.log(idx) }
VM537:2 0
VM537:2 1
VM537:2 2
VM537:2 last
undefined

```
Might have stuff to do with with the shims we're using.

**Comment by p3drosola** [comment]
that's definitely not normal javascript behaviour... and not an issue VirtualCollection should deal with.

**Comment by jacobbuck** [comment]
`for-in` will loop through items in the prototype too. Normally you should use `hasOwnProperty()` to prevent this behaviour.

I've fixed this in #67 by using `_.invoke` instead.
